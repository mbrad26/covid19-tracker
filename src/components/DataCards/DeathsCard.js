import React from 'react';
import { Card } from 'react-bootstrap';
import CountUp from 'react-countup';

import ChartComponent from '../Chart/ChartComponent.js';

const DeathsCard = ({
  deaths,
  country,
  lastUpdated,
  deathsIncrease,
  historicalData
  }) => {
  const labels = historicalData && Object.keys(historicalData);
  const values = historicalData && Object.values(historicalData);

  return (
    <Card>
      <Card.Body>
        <Card.Title>{country}: <span>
          <CountUp
            prefix='+'
            start={0}
            end={deathsIncrease}
            duration={2.5}
            separator=','
          />
          </span>
          <i className="fas fa-skull-crossbones"></i>
        </Card.Title>
        <ChartComponent
          labels={labels}
          values={values}
          label={'Deaths'}
          color={'#bd0026'}
        />
        <Card.Text>
          {deaths.toLocaleString()}
          <span className='text-muted' id='update'>{lastUpdated}</span>
        </Card.Text>
      </Card.Body>
    </Card>
  );
};

export default DeathsCard;
