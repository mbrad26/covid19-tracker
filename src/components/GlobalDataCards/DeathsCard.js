import React from 'react';
import { Card } from 'react-bootstrap';

const DeathsCard = ({ data }) => (
  <Card className='shadow' id='danger'>
    <Card.Body>
      <Card.Title>Deaths</Card.Title>
      <Card.Text>
        {data.toLocaleString()}
      </Card.Text>
    </Card.Body>
  </Card>
)

export default DeathsCard;
