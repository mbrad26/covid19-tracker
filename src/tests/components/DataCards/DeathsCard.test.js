import React from 'react';
// import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import { globalData, historicalData } from '../../fixtures';
import DeathsCard from '../../../components/DataCards/DeathsCard';

describe('DeathsCard', () => {
  it('renderes snapshot', () => {
    const { container } = render(
      <DeathsCard
        deaths={globalData.deaths}
        country={globalData.country ? globalData.country : 'Worldwide'}
        lastUpdated={new Date(globalData.updated).toDateString()}
        deathsIncrease={globalData.todayDeaths}
        historicalData={historicalData}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
