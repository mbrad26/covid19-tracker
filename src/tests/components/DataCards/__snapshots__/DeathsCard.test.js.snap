// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeathsCard renderes snapshot 1`] = `
<div
  class="card"
>
  <div
    class="card-body"
  >
    <div
      class="card-title h5"
    >
      Worldwide
      : 
      <span>
        <span>
          +0
        </span>
      </span>
      <i
        class="fas fa-skull-crossbones"
      />
    </div>
    <canvas
      class="chartjs-render-monitor"
      height="0"
      style="display: block;"
      width="0"
    />
    <p
      class="card-text"
    >
      855,941
      <span
        class="text-muted"
        id="update"
      >
        Tue Sep 01 2020
      </span>
    </p>
  </div>
</div>
`;
